eureka/S3_data_reduction/niriss.py

```diff
@@ -460,53 +187,67 @@ def wave_NIRISS(wavefile, meta):
        The name of the .FITS file with the wavelength
        solution.
     meta : object
+    filename : str, optional
+       The flux filename. Default is None. Needs a filename if
+       the `meta` class is not provided.
 
     Returns
     -------
     meta : object
     """
-    hdu = fits.open(wavefile)
-
-    meta.wavelength_order1 = hdu[1].data + 0.0
-    meta.wavelength_order2 = hdu[2].data + 0.0
-    meta.wavelength_order3 = hdu[3].data + 0.0
-
-    hdu.close()
+    if meta is not None:
+        rampfitting_results = datamodels.open(meta.filename)
+    else:
+        rampfitting_results = datamodels.open(filename)
 
-    return meta
+    # Run assignwcs step on Stage 1 outputs:
+    assign_wcs_results = calwebb_spec2.assign_wcs_step.AssignWcsStep.call(rampfitting_results)
 
-def flag_bg(data, meta):
-    '''Outlier rejection of sky background along time axis.
-
-    Parameters
-    ----------
-    data:   DataClass
-        The data object in which the fits data will stored
-    meta:   MetaClass
-        The metadata object
-
-    Returns
-    -------
-    data:   DataClass
-        The updated data object with outlier background pixels flagged.
-    '''
-
-    print('WARNING, niriss.flag_bg is not yet implemented!')
+    # Extract 2D wavelenght map for order 1:
+    rows, columns = assign_wcs_results.data[0,:,:].shape
+    wavelength_map = np.zeros([3, rows, columns])
+    
+    # Loops through the three orders to retrieve the wavelength maps
+    if orders is None:
+        orders = [1,2,3]
+
+    for order in orders:
+        for row in tqdm(range(rows)):
+            for column in range(columns):
+                wavelength_map[order-1, row, column] = assign_wcs_results.meta.wcs(column, 
+                                                                                   row, 
+                                                                                   order)[-1]
+    if inclass == False:
+        meta.wavelength_order = wavelength_map
+        return meta
+    else:
+        return wavelength_map
+
+
+def flag_bg(data, meta, readnoise=11, sigclip=[4,4,4], 
+            box=(5,2), filter_size=(2,2), bkg_estimator=['median'], isplots=0):
+    """ 
+    I think this is just a wrapper for fit_bg, because I perform outlier
+    flagging at the same time as the background fitting.
+    """
+    data, bkg, bkg_var = fit_bg(data, meta, readnoise, sigclip, 
```

Comment:
These two steps (flag_bg and fit_bg) should be separate in order to coincide with the other instruments.  The latter needs to happen in `bg.BGsubtraction()`.

Reply:
This is fine for now as NIRISS has its own S3 reduction file. will need to reassess in next PR